networkx/algorithms/tests/test_euler.py

```diff
@@ -150,21 +150,17 @@ def test_has_eulerian_path_isolated_node(self):
         G.add_node(3)
         assert nx.has_eulerian_path(G)
 
-    def test_has_eulerian_path_not_weakly_connected(self):
-        G = nx.DiGraph()
-        H = nx.Graph()
+    @pytest.mark.parametrize("graph", (nx.Graph, nx.DiGraph))
+    def test_has_eulerian_path_not_weakly_connected(self, graph):
+        G = graph()
```

Comment:
```suggestion
    @pytest.mark.parametrize("G", (nx.Graph(), nx.DiGraph()))
    def test_has_eulerian_path_not_weakly_connected(self, G):
```

Looking at this again, turns out my previous suggestion could be made even shorter by passing in instances rather than classes for the parameter. Not a huge improvement, but saves a line :)